cloudferrylib/os/actions/transport_ephemeral.py

```diff
@@ -185,3 +188,19 @@ def copy_ephemeral_iscsi_to_ceph(self, src_cloud, info):
             inst[EPHEMERAL][PATH_SRC] = path_src_temp_raw
 
         transporter.run(info=info)
+
+    @staticmethod
+    def rebase_diff(dst_cloud, info):
+        for instance_id, obj in info[utl.INSTANCES_TYPE].items():
+            image_id = obj['instance']['image_id']
+            new_backing_file = hashlib.sha1(image_id).hexdigest()
+            diff = obj['diff']
+            host = diff['host_dst']
+            qemu_img = qemu_img_util.QemuImg(dst_cloud.config.dst,
+                                             dst_cloud.config.migrate,
+                                             host)
+            diff_path = diff['path_dst']
+            backing_path = qemu_img.detect_backing_file(diff_path, None)
+            backing_dir, _ = backing_path.rsplit('/', 1)
+            new_backing_path = backing_dir + '/' + new_backing_file
```

Comment:
I know that in theory this could be run on windows machine, but still seem to be safer with `os.path.join` and `os.path.basename`.